.github/workflows/deploy-docker-compose.yml

```diff
@@ -0,0 +1,161 @@
+name: Deploy Docker Compose
+
+on:
+  workflow_call:
+    inputs:
+      environment:
+        type: string
+        description: "The deployment environment (e.g., production, staging)"
+        required: true
+      docker-compose-file:
+        type: string
+        default: "./docker-compose.yml"
```

Comment:
Small thing but wanted to mention in here
The preferred compose file name is [compose.yaml](https://docs.docker.com/compose/intro/compose-application-model/#the-compose-file)
If I remember correctly `docker-compose.yml` and also command `docker-compose` is an old convention for docker V1